File: src/main/java/tk/avabin/tdg/beans/Entities/Character.java
package tk.avabin.tdg.beans.Entities;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import javax.persistence.*;
import java.util.Set;

/**
 * Created by Avabin on 13.03.2017.
 */
@Component
@Scope("prototype")
@Data
@NoArgsConstructor
@Entity
@Table(name = "game_character")
public class Character {
    @Id
    @GeneratedValue
    @Column(name = "character_id", nullable = false, unique = true)
    private int id;
    @Column(name = "character_name", nullable = false, unique = true)
    private String name;
    @ManyToOne
    private User owner;
    @OneToMany
    private Set<RPGClassAndLevel> classesAndLevels;
    private String race;
    private String alignment;
    private String deity;
    private String size;
    private Integer age;
    private String gender;
    private Integer height;
    private Integer weight;
    private Integer eyes;
    private Integer hair;
    private Integer skin;
    @Column(name = "hit_dice", length = 5)
    private String hitDice;
    private Integer hitPoints;
    private Integer initiative;
    private Integer speed;
    @Column(name = "base_attack_bonus")
    private String baseAttackBonus;
    @Column(name = "spell_resistance")
    private Integer spellResistance;
    @Column(name = "armor_class")
    private Integer armorClass;
    private Integer strength;
    private Integer dexterity;
    private Integer constitution;
    private Integer intelligence;
    private Integer wisdom;
    private Integer charisma;
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Item> items;
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Spell> spells;
    private Integer rank0SpellsPerDay;
    private Integer rank1SpellsPerDay;
    private Integer rank2SpellsPerDay;
    private Integer rank3SpellsPerDay;
    private Integer rank4SpellsPerDay;
    private Integer rank5SpellsPerDay;
    private Integer rank6SpellsPerDay;
    private Integer rank7SpellsPerDay;
    private Integer rank8SpellsPerDay;
    private Integer rank9SpellsPerDay;
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Skill> skills;
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Feat> feats;
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Language> languages;
}

File: src/main/kotlin/tk/avabin/tdg/beans/entities/Character.kt
package tk.avabin.tdg.beans.entities

import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.persistence.Table

/**
 * Created by Avabin on 13.03.2017.
 */
@Component
@Scope("prototype")
@Entity
@Table(name = "game_character")
data class Character(
        @Id
        @GeneratedValue
        @Column(name = "character_id", nullable = false, unique = true)
        var id: Int = 0,
        @Column(name = "character_name", nullable = false, unique = true)
        var name: String = "",
        @ManyToOne
        var owner: User? = null,
        @OneToMany(cascade = arrayOf(CascadeType.ALL))
        var classesAndLevels: Set<RPGClassAndLevel>? = HashSet(),
        var race: String = "",
        var alignment: String = "",
        var deity: String = "",
        var size: String = "",
        var age: Int = 0,
        var gender: String = "",
        var height: Int = 0,
        var weight: Int = 0,
        var eyes: Int = 0,
        var hair: Int = 0,
        var skin: Int = 0,
        @Column(name = "hit_dice", length = 5)
        var hitDice: String = "",
        var hitPoints: Int = 0,
        var initiative: Int = 0,
        var speed: Int = 0,
        @Column(name = "base_attack_bonus")
        var baseAttackBonus: String = "",
        @Column(name = "spell_resistance")
        var spellResistance: Int = 0,
        @Column(name = "armor_class")
        var armorClass: Int = 0,
        var strength: Int = 0,
        var dexterity: Int = 0,
        var constitution: Int = 0,
        var intelligence: Int = 0,
        var wisdom: Int = 0,
        var charisma: Int = 0,
        @OneToMany(fetch = FetchType.EAGER, cascade = arrayOf(CascadeType.ALL))
        var items: Set<Item> = HashSet(),
        @OneToMany(fetch = FetchType.EAGER, cascade = arrayOf(CascadeType.ALL))
        var spells: Set<Spell> = HashSet(),
        var rank0SpellsPerDay: Int = 0,
        var rank1SpellsPerDay: Int = 0,
        var rank2SpellsPerDay: Int = 0,
        var rank3SpellsPerDay: Int = 0,
        var rank4SpellsPerDay: Int = 0,
        var rank5SpellsPerDay: Int = 0,
        var rank6SpellsPerDay: Int = 0,
        var rank7SpellsPerDay: Int = 0,
        var rank8SpellsPerDay: Int = 0,
        var rank9SpellsPerDay: Int = 0,
        @OneToMany(fetch = FetchType.EAGER, cascade = arrayOf(CascadeType.ALL))
        var skills: Set<Skill> = HashSet(),
        @OneToMany(fetch = FetchType.EAGER, cascade = arrayOf(CascadeType.ALL))
        var feats: Set<Feat> = HashSet(),
        @OneToMany(fetch = FetchType.EAGER, cascade = arrayOf(CascadeType.ALL))
        var languages: Set<Language> = HashSet()
)